trpcultivate_phenotypes/tests/src/Kernel/Forms/ConfigOntologyTermsFormExceptionsTest.php

```diff
@@ -0,0 +1,121 @@
+  /**
+   * Modules to enable.
+   *
+   * @var array
+   */
+  protected static $modules = [
+    'tripal',
+    'tripal_chado',
+    'trpcultivate_phenotypes',
+  ];
+
+  /**
+   * Class instance of ontology settings form.
+   *
+   * @var \Drupal\trpcultivate_phenotypes\Form\TripalCultivatePhenotypesOntologySettingsForm
+   */
+  protected $ontology_form;
+
+  /**
+   * Configuration.
+   *
+   * @var \Drupal\Core\Config\Config
+   */
+  private $config;
+
+  /**
+   * A Database query interface for querying Chado using Tripal DBX.
+   *
+   * @var \Drupal\tripal_chado\Database\ChadoConnection
+   */
+  protected ChadoConnection $chado_connection;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+
+    // Set test environment.
+    \Drupal::state()->set('is_a_test_environment', TRUE);
+
+    // Create a test chado instance as needed by our service.
+    $this->chado_connection = $this->createTestSchema(ChadoTestKernelBase::PREPARE_TEST_CHADO);
+
+    // Insert a genus to the database.
+    $this->chado_connection->insert('1:organism')
+      ->fields(['genus', 'species'])
+      ->values([
+        'genus' => 'Lens',
+        'species' => 'databasica',
+      ])
+      ->execute();
+
+    $this->container = \Drupal::getContainer();
+
+    // Mock the getGenusOntologyConfigValues method on genus ontology service
+    // so it will return a string instead of null or an array.
+    $mock_ontology_service = $this->getMockBuilder(TripalCultivatePhenotypesGenusOntologyService::class)
+      ->setConstructorArgs([$this->container->get('config.factory'),
+        $this->container->get('tripal_chado.database'),
+        $this->container->get('tripal.logger'),
+      ])
+      ->onlyMethods(['getGenusOntologyConfigValues'])
+      ->getMock();
+
+    $mock_ontology_service->method('getGenusOntologyConfigValues')
+      ->willReturn('someValue');
+
+    $this->container->set('trpcultivate_phenotypes.genus_ontology', $mock_ontology_service);
+
+    // Install module configuration.
+    $this->installConfig(['trpcultivate_phenotypes']);
+
+    // Create new instance of the TripalCultivatePhenotypesOntologySettingsForm.
+    $this->ontology_form = TripalCultivatePhenotypesOntologySettingsForm::create($this->container);
+  }
+
+  /**
+   * Tests the buildForm method for the exception cases.
+   */
+  public function testBuildFormWithExceptions() {
+    $form = [];
+    $form_state = new FormState();
+
+    $service_terms = $this->container->get('trpcultivate_phenotypes.terms');
+    $service_terms->loadTerms();
+
+    // Test for the case where the config value is not an array or null.
+    $this->ontology_form->buildForm($form, $form_state);
+
+    $errors = \Drupal::messenger()->messagesByType('error');
+    $this->assertCount(1, $errors,
+      'We expect an error message when config value for a genus is not an array, but it was not thrown.');
+
```

Comment:
```suggestion

$this->assertStringContainsString(
      'We were unable to retrieve configuration values for the lens Genus.',
      (string) $errors[0],
      'The error message does not match expected error message if settings form could not retrieve values for the genus.'
    );

```

To confirm that the error message is the expected message.

Reply:
This comment has been resolved